src/cxy_cache.erl

```diff
@@ -363,8 +363,10 @@ fetch_item(Cache_Name, Key) ->
 %% Fetch from Generation 1 or 2, after updating with a newer key version or leaving existing newest key version.
 refresh_item(Cache_Name, Key) ->
     Found_Fn     = fun(Fn_Cache_Name, _Value, _Version, Fn_New_Gen_Id, Fn_Mod, Fn_Key) ->
-                           Fn_New_Cached_Value = create_new_value(Fn_Cache_Name, Fn_New_Gen_Id, Fn_Mod,Fn_Key),
-                           return_cache_refresh(Fn_Cache_Name, Fn_New_Cached_Value)
+                           case create_new_value(Fn_Cache_Name, Fn_New_Gen_Id, Fn_Mod,Fn_Key) of
+                               no_value_available                  -> return_cache_refresh(Fn_Cache_Name, no_value_available);
+                               {_New_Version, Fn_New_Cached_Value} -> return_cache_refresh(Fn_Cache_Name, Fn_New_Cached_Value)
```

Comment:
It started as a bare value. When we versioned, we made a pair {Version, Value} or no_value_available so that there **is** a difference between a value and the missing value signifier. I ran dialyzer and checks things out. Are you referring to the fact that a record might have types declared for each field? They are both any() because I don't place constraints on what constitutes a value or a version for the caller. The behaviour allows a Mod:is_later_version/2 if > is not sufficient. But the record would be ``-define(value_pair, {version :: any(), value :: any()}.

If you look at what is stored in the ets cache you will see `` #cxy_cache_value{key=Key, version=Insert_Vsn, value=Insert_Value} = Obj`` as used in functions further down.